pkg/openstack/ovn.go

```diff
@@ -28,6 +25,14 @@ func ReconcileOVN(ctx context.Context, instance *corev1beta1.OpenStackControlPla
 			},
 		}
 
+		if !instance.Spec.Ovn.Enabled {
+			if res, err := CheckDeleteDisabled(ctx, helper, OVNDBCluster); err != nil {
+				return res, err
+			}
+			instance.Status.Conditions.Remove(corev1beta1.OpenStackControlPlaneOVNReadyCondition)
+			return ctrl.Result{}, nil
```

Comment:
This should be `continue` instead of a `return`.  Otherwise other `OVNDBCluster` and `OVNNorthD` resources will not be deleted when OVN is disabled.

Reply:
Nice catch, thanks!